lib/find-python.js

```diff
@@ -1,329 +1,710 @@
+   * Python executable has the correct version to use.
+   *
+   * @private
+   * @argument {string} execPath path to check
+   * @returns {Promise}
+   */
+  checkExecPath (execPath) {
+    // Returning new Promise instead of forwarding existing
+    // to pass both path and version
+    return new Promise((resolve, reject) => {
+      this.log.verbose(`executing "${colorizeOutput(colorHighlight.GREEN, execPath)}" to get version`)
+
+      // do not wrap with quotes because executing without shell
+      this.run(execPath, this.argsVersion, false)
+        .then((ver) => {
+          this.log.silly(colorizeOutput(colorHighlight.GREEN, 'version got:'), ver)
+          // ? may be better code for version check
+          // ? may be move log messages to catchError func
+          const range = new semver.Range(this.semverRange)
+          let valid = false
+
+          try {
+            valid = range.test(ver)
+            // throw new Error("test error")
+          } catch (err) {
+            this.log.silly(`range.test() threw:\n${err.stack}`)
+            this.addLog(
+              `"${colorizeOutput(colorHighlight.RED, execPath)}" does not have a valid version`
+            )
+            this.addLog('Is it a Python executable?')
+
+            // if you need to pass additional data, use ErrorWithData class
+            // you can also use any Error capable object
+            return reject(err)
+          }
+
+          if (!valid) {
+            this.addLog(
+              `version is ${colorizeOutput(
+                colorHighlight.RED,
+                ver
+              )} - should be ${colorizeOutput(colorHighlight.RED, this.semverRange)}`
+            )
+            this.addLog(
+              colorizeOutput(colorHighlight.RED, 'THIS VERSION OF PYTHON IS NOT SUPPORTED')
+            )
+            // if you need to pass additional data, use ErrorWithData class
+            reject(new Error(`Found unsupported Python version ${ver}`))
+          }
+
+          resolve({ path: execPath, version: ver })
+        })
+        .catch(reject)
+    })
+  }
+
+  /**
+   * Run an executable or shell command, trimming the output.
+   *
+   * @private
+   * @argument {string} exec command or path without arguments to execute
+   * @argument {string[]} args command args
+   * @argument {boolean} shell need be documented
+   * @returns {Promise}
+   */
+  run (exec, args, shell) {
+    return new Promise(
+      /**
+       * @this {PythonFinder}
+       * @argument {function} resolve
+       * @argument {function} reject
+       */
+      function (resolve, reject) {
+        const env = extend({}, this.env)
+        env.TERM = 'dumb'
+        /** @type {cp.ExecFileOptions} */
+        const opts = { env: env, shell: shell }
+
+        this.log.verbose(`${colorizeOutput(colorHighlight.GREEN, 'execFile')}: exec = `, exec)
+        this.log.verbose(`${colorizeOutput(colorHighlight.GREEN, 'execFile')}: args = `, args)

```

Comment:
These two logs used to be loglevel `silly`.

<details>

Example (before):

```
gyp sill find Python execFile: exec = "python3.5"
gyp sill find Python execFile: args = ["-c","import sys; print(sys.executable);"]
```

Now (verbose):

```
gyp verb find Python execFile: exec =  python3.5
gyp verb find Python execFile: args =  [ '/home/[user]/node-gyp/lib/find-python-script.py' ]
```

</details>

I personally prefer them to remain `silly` level, which is one level more obscure than `verbose`, and keeps the error output cleaner and more to-the-point. I peronally don't think end-users need to know which Python script is being executed to test the Python version.

But again, I am not an official maintainer here, and their opinions matter more than mine. I would highly recommend waiting for an actual maintainer to review this before making changes based on my feedback. (I believe they are all quite busy with the imminent Node v16 promotion to be active Node release. That happens on 20th April 2021, in five days.)

Reply:
Thank you for feedback)
I think we should define which log level is for what. 

In general, i agree with you that report about values of vars should be on `silly` level. But may be it would be more useful to provide data about critical vars to `verbose` level. In fact fail message of node-gyp doesn't tell how to turn `silly` level. It advises to add `--verbose` i.e. turn on `verbose` not `silly` level. I think these two vars contain fairly critical data for devs (may idicate when wrong data is passed to `run` function). As i can see, openning an issue require to provide `verbose` output. Ask to user to provide `silly` logs may cost time. Above is my point of view)

Reply:
Even more, i think we should put `execFile result: stdout =` to verbose level too. Devs should see, what the script actually get from `execFile`